File: lib/java/src/main/java/io/github/brenbar/uapi/internal/GenerateRandomBoolean.java
package io.github.brenbar.uapi.internal;

import io.github.brenbar.uapi.RandomGenerator;

public class GenerateRandomBoolean {
    static Object generateRandomBoolean(Object blueprintValue, boolean useBlueprintValue,
            RandomGenerator randomGenerator) {
        if (useBlueprintValue) {
            return blueprintValue;
        } else {
            return randomGenerator.nextBoolean();
        }
    }
}

File: lib/go/internal/types/GenerateRandomBoolean.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// GenerateRandomBoolean returns a pseudo-random boolean, optionally reusing a blueprint value.
func GenerateRandomBoolean(blueprintValue any, useBlueprintValue bool, ctx *GenerateContext) any {
	if useBlueprintValue {
		return blueprintValue
	}
	if ctx == nil || ctx.RandomGenerator == nil {
		return nil
	}
	return ctx.RandomGenerator.NextBoolean()
}
